### PR TITLE
Replacing ccr during config file generation

### LINKED ISSUE #990: [TC-353] Can't use CNAME hostname with "ccr" prefix
<p>If you put a HOST_REGEX of "ccr.example.kabletown.net", it will expand to the server hostname like "edge01.example.kabletown.net". This makes the remap.config to have the wrong hostname. This might not affect anything but it should be changed to something unrelated to a specific prefix.</p>

<p>The code ConfigFile.pm is causing the current behavior :</p>

<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java">
				my $hname = $ds_type =~ /^DNS/ ? <span class="code-quote">"edge"</span> : <span class="code-quote">"ccr"</span>;
				my $portstr = "";
...
	$map_from =~ s/ccr/$host_name/;
</pre>
</div></div>

Author: Steve Malenfant
JIRA Link: <a href="https://issues.apache.org/jira/browse/TC-353">https://issues.apache.org/jira/browse/TC-353</a>
Found Version: 2.1.0

### PR DESCRIPTION
**Pull request by smalenfant**
- Allow CNAME for hostname with `ccr` by replacing the regex
- Close #990

### CODE REVIEW

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-PR/939/
Test PASSed.

File: traffic_ops/app/lib/API/Configs/ApacheTrafficServer.pm [hunk 1]
```diff
@@ -915,9 +915,9 @@ sub ds_data {
 				my $re = $host_re;
 				$re =~ s/\\//g;
 				$re =~ s/\.\*//g;
-				my $hname = $ds_type =~ /^DNS/ ? $dsinfo->routing_name : "ccr";
+				my $hname = $ds_type =~ /^DNS/ ? $dsinfo->routing_name : "__http__";
 				my $portstr = "";
-				if ( $hname eq "ccr" && $server_obj->tcp_port > 0 && $server_obj->tcp_port != 80 ) {
+				if ( $hname eq "__http__r" && $server_obj->tcp_port > 0 && $server_obj->tcp_port != 80 ) {
```
Comment (mitchell852) [inline traffic_ops/app/lib/API/Configs/ApacheTrafficServer.pm]: @smalenfant - this looks like a typo? __http__r?

**Comment by rawlinp** [comment]
What about traffic_ops/app/lib/UI/ConfigFiles.pm? I think you had that in your original PR but missed it in this one?

**Comment by smalenfant** [comment]
@mitchell852 I'll fix that.

@rawlinp I can put `traffic_ops/app/lib/UI/ConfigFiles.pm` back in. I wasn't sure that we were still using this (but might be necessary for backward compatibility for ORT/Ops).

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-PR/978/
Test PASSed.

**Comment by rawlinp** [comment]
@smalenfant yeah, I think until we officially deprecate the TO UI we should try to keep parity.

This looks good to me 👍

**Pull request merged**